From 3a1abb555b844c79a80615727b11019403767ce7 Mon Sep 17 00:00:00 2001
From: Jamie Mansfield <jmansfield@cadixdev.org>
Date: Mon, 5 Apr 2021 21:10:37 +0100
Subject: [PATCH] GH-3575 Fix build on Fedora 34
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Presumably this is caused by the bump to GCC 11 in Fedora 34. See
the error that did occur below...

    ./MultiMC5/application/KonamiCode.cpp: In member function ‘void KonamiCode::input(QEvent*)’:
    ./MultiMC5/application/KonamiCode.cpp:38:23: error: comparison of integer expressions of different signedness: ‘int’ and ‘std::array<Qt::Key, 10>::size_type’ {aka ‘long unsigned int’} [-Werror=sign-compare]
       38 |         if(m_progress == konamiCode.size())
          |            ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
---
 application/KonamiCode.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/application/KonamiCode.cpp b/application/KonamiCode.cpp
index 4c5af8371..46a2a0b2e 100644
--- a/application/KonamiCode.cpp
+++ b/application/KonamiCode.cpp
@@ -35,7 +35,7 @@ void KonamiCode::input(QEvent* event)
         {
             m_progress = 0;
         }
-        if(m_progress == konamiCode.size())
+        if(m_progress == static_cast<int>(konamiCode.size()))
         {
             m_progress = 0;
             emit triggered();
